src/components/chart/buggy-basic-chart.js

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import _ from 'underscore'
+import './chart.css';
+import Cardiogram from '../../models/cardiogram';
+
+const REFRESH_INTERVAL_MS = 30 * 1000;  // refresh every 30 seconds
+
+/**
+ *  A React component which displays a heart rate bar chart, with time on the X axis and heart rate
+ *  on the Y axis.
+ *
+ *  TODO(brandon): fix all these darn bugs!
+ */
+export default class BuggyHeartRateChart extends React.Component {
+  static propTypes() {
+    return {
+      cardiogram: React.PropTypes.instanceOf(Cardiogram).isRequired
+    };
+  }
+
+  constructor(props) {
+    super(props);
+    this.svgWidthPx = 300;
+    this.svgHeightPx = 200;
+    this.minTimestamp = _.min(_.pluck(props.cardiogram.data, 'start'));
+    this.maxTimestamp = _.max(_.pluck(props.cardiogram.data, 'start'));
+  }
+
+  componentDidMount() {
+    // refresh cardiogram data from the server every 30 seconds.
+    setInterval(this.refreshCardiogramData, REFRESH_INTERVAL_MS);
+  }
+
+  timestampToX(timestamp) {
+    return this.svgWidthPx * (timestamp - this.minTimestamp) / (this.minTimestamp - this.maxTimestamp);
+  }
+
+  heartRateToY(heartRate) {...}  // implementation elided for brevity
+
+  refreshCardiogramData() {
+    this.sendJSONRequestToServer.then(function(responseJson) {
+      this.props.cardiogram = responseJson.cardiogram;
+    })
+  }
+
+  // returns a promise. Implementation elided for brevity.
+  sendJSONRequestToServer() {...}
+
+  render() {
+    return (
+      <div
+        class="chart"
+        onClick={this.refreshCardiogramData}
+      >
+
+        <svg height={this.svgHeightPx} width={this.svgWidthPx}>
+          {
+            _.map(this.props.cardiogram.data, (b) => {
+              const y = this.heartRateToY(b.value);
+              return <rect
```

Comment:
Good practice to add parentheses for return or you may run into https://github.com/Cardiogram/Cardiogram/pull/4490